File: benchmark/js/utils/http.js
const http = require('http');

exports.get = (url, cb) => {
    // Server-Side Request Forgery [CWE-918]
    http.get(url, (res) => {
        let data = '';
        res.on('data', chunk => data += chunk);
        res.on('end', () => cb(data));
    }).on('error', () => cb('Error'));
};

File: benchmark/go/services/http.go
package services

import (
	"io"
	"net/http"
)

func FetchHttp(url string) string {
	// Server-Side Request Forgery (SSRF) [CWE-918]
	resp, err := http.Get(url)
	if err != nil {
		return ""
	}
	defer resp.Body.Close()
	body, _ := io.ReadAll(resp.Body)
	return string(body)
}
